compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -204,6 +216,80 @@ impl<'cx, 'tcx> TypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
                     }
                 }
             }
+            ty::Projection(data) if !data.trait_ref(self.infcx.tcx).has_escaping_bound_vars() => {
+                // See note in `rustc_trait_selection::traits::project`
+
+                // One other point mentioning: In `traits::project`, if a
+                // projection can't be normalized, we return an inference variable
+                // and register an obligation to later resolve that. Here, the query
+                // will just return ambiguity. In both cases, the effect is the same: we only want
+                // to return `ty` because there are bound vars that we aren't yet handling in a more
+                // complete way.
+
+                let tcx = self.infcx.tcx;
+                let infcx = self.infcx;
+                let replaced = crate::traits::project::BoundVarReplacer::replace_bound_vars(
+                    infcx,
+                    &mut self.universes,
+                    data,
+                );
+                let (data, mapped_regions, mapped_types, mapped_consts) = match replaced {
+                    Some(r) => r,
+                    None => {
+                        bug!("{:?} {:?}", data, self.universes);
```

Comment:
Oh...hmm. interesting. Didn't mean to leave this. But since all the tests pass, maybe the only place that we pass escaping bound vars here is under polymorphization (and supposedly there's no test for the intersection of polymorphization and GATs). But the underlying problem still exists (normalize_erasing_regions). So not sure if it's better to just ICE if we get passed bound vars (gives us test cases) or to fail and return an error (which might still allow us to succeed compilation overall).

Thoughts @nikomatsakis?